addons/website_sale/models/sale_order.py

```diff
@@ -32,7 +32,7 @@ class SaleOrder(models.Model):
                  'payment_acquirer_id', 'payment_acquirer_id.provider')
     def _compute_can_directly_mark_as_paid(self):
         for order in self:
-            order.can_directly_mark_as_paid = order.state in ['sent', 'sale'] and order.payment_tx_id.state != 'done' and order.payment_acquirer_id.provider in ['transfer', 'manual']
+            order.can_directly_mark_as_paid = order.state in ['sent', 'sale'] and order.payment_tx_id and order.payment_acquirer_id.provider in ['transfer', 'manual']
```

Comment:
To put directly in https://github.com/odoo/odoo/pull/17067/commits/83280a288e799aad565849c652b924cf821cbc88to have only one version of the method.